src/host/ft_host/API_FillOutputTests.cpp

```diff
@@ -58,4 +58,140 @@ class FillOutputTests
                                                                 &charsWritten));
         VERIFY_ARE_EQUAL(1u, charsWritten);
     }
+
+    TEST_METHOD(UnsetWrap)
+    {
+        // WARNING: If this test suddenly decides to start failing,
+        // this is because the wrap registry key is not set.
+        // TODO GH #2859: Get/Set Registry Key for Wrap
+
+        HANDLE hConsole = GetStdOutputHandle();
+        DWORD charsWritten = 0;
+
+        CONSOLE_SCREEN_BUFFER_INFOEX sbiex = { 0 };
+        sbiex.cbSize = sizeof(sbiex);
+        GetConsoleScreenBufferInfoEx(hConsole, &sbiex);
+
+        const auto ogConsoleWidth = sbiex.dwSize.X;
+
+        std::wstring input = L"";
+        for (SHORT i = 0; i < ogConsoleWidth + 2; i++)
+        {
+            input.append(L"a");
+        }
+
+        // Write until a wrap occurs
+        VERIFY_WIN32_BOOL_SUCCEEDED(WriteConsoleW(hConsole,
+                                                  input.data(),
+                                                  gsl::narrow_cast<DWORD>(input.size()),
+                                                  &charsWritten,
+                                                  nullptr));
+
+        // Verify wrap occurred
+        std::unique_ptr<wchar_t[]> bufferText = std::make_unique<wchar_t[]>(ogConsoleWidth);
+        DWORD readSize = 0;
+        VERIFY_WIN32_BOOL_SUCCEEDED(ReadConsoleOutputCharacterW(hConsole,
+                                                                bufferText.get(),
+                                                                ogConsoleWidth,
+                                                                { 0, 0 },
+                                                                &readSize));
+
+        for (DWORD i = 0; i < readSize; i++)

```

Comment:
Man.... can you do some sort of string compare here instead?

`WEX::Common::String()` has a bunch of helpers for allocating two strings that are comparable in `VERIFY_ARE_EQUAL` and show up in the logs correctly.